pkg/api/customization/workload/workload.go

```diff
@@ -36,33 +38,108 @@ func (a ActionWrapper) ActionHandler(actionName string, action *types.Action, ap
 	namespace, name := splitID(deployment.ID)
 	switch actionName {
 	case "rollback":
-		actionInput, err := parse.ReadBody(apiContext.Request)
+		input, err := handler.ParseAndValidateActionBody(apiContext, apiContext.Schemas.Schema(&projectschema.Version,
```

Comment:
can we extract rollback code to a separate method? it would be better from visibility perspective